apps/glusterfs/block_volume_entry_create.go

```diff
@@ -56,9 +56,16 @@ func (v *BlockVolumeEntry) createBlockVolumeRequest(db *bolt.DB,
 			return err
 		}
 
-		v.Info.BlockVolume.Hosts = bhvol.Info.Mount.GlusterFS.Hosts
-		v.Info.Hacount = len(v.Info.BlockVolume.Hosts)
+		if v.Info.Hacount > 0 {
+			for i := 0; i < v.Info.Hacount; i++ {
+				v.Info.BlockVolume.Hosts = append(v.Info.BlockVolume.Hosts, bhvol.Info.Mount.GlusterFS.Hosts[i])
```

Comment:
@MohamedAshiqrh iic, mount hosts are brick peers and NOT TSP peers. If yes, we are sending a wrong list. We should send TSP list instead of brick peers list.